vkcs/kubernetes/shared.go

```diff
@@ -103,3 +103,59 @@ func kubernetesStateRefreshFunc(client *gophercloud.ServiceClient, clusterID str
 		return c, c.NewStatus, nil
 	}
 }
+
+func kubernetesNodeGroupCreateStateRefreshFunc(client *gophercloud.ServiceClient, nodeGroupID string, clusterID string) retry.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		n, err := nodegroups.Get(client, nodeGroupID).Extract()
+		if err != nil {
+			if errutil.IsNotFound(err) {
+				return n, string(nodeGroupStatusCreating), nil
+			}
+			return nil, "", err
+		}
+		c, err := clusters.Get(client, clusterID).Extract()
+		if err != nil {
+			if errutil.IsNotFound(err) {
+				return c, string(clusterStatusNotFound), nil
```

Comment:
It looks like a very strange case, so I would handle it the same way as other errors.

Reply:
Rewrote to use separate wait function for node group, thus cluster wait function will remain same